### PR TITLE
Add UI support for Seeed Studio XIAO ESP32C3

### PR DESCRIPTION
**Pull request by lakshanthad**
Hello,

Please help to merge the Seeed Studio XIAO ESP32C3 board support to ESPHome Dashboard.

Here is board support on PlatformIO for reference:
https://github.com/platformio/platform-espressif32/blob/develop/boards/seeed_xiao_esp32c3.json

Thank you.

### CODE REVIEW

File: src/wizard/boards.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export const boardSelectOptions = html`
     <option value="lolin_d32_pro">Wemos Lolin D32 Pro</option>
     <option value="featheresp32">Adafruit ESP32 Feather</option>
     <option value="m5stack-core-esp32">M5Stack Core ESP32</option>
+    <option value="seeed_xiao_esp32c3">Seeed Studio XIAO ESP32C3</option>
```
Comment (balloob) [inline src/wizard/boards.ts:11]: Can you add it to the "Other ESP32s" section.

Comment (lakshanthad) [reply src/wizard/boards.ts:11]: Can I add to the start of "Other ESP32s" section?

Comment (lakshanthad) [reply src/wizard/boards.ts:11]: Also, will it use the correct "variant" as "esp32c3"?

Comment (balloob) [reply src/wizard/boards.ts:11]: > Can I add to the start of "Other ESP32s" section?

We're not a marketing tool. Let's sort this alphabetically.

Maybe we should break out the C3.

Comment (lakshanthad) [reply src/wizard/boards.ts:11]: > Maybe we should break out the C3.

Do you mean to have a new section for ESP32C3?

It seems there is a number of ESP32C3 boards available already.

Comment (balloob) [reply src/wizard/boards.ts:11]: Yeah that would probably make more sense.

Comment (lakshanthad) [reply src/wizard/boards.ts:11]: Hi. Sorry for late response.

I have added all the ESP32C3 boards under "ESP32C3" section

File: src/wizard/boards.ts [hunk 2]
```diff
@@ -21,6 +21,18 @@ export const boardSelectOptions = html`
     <option value="thing">Sparkfun ESP8266 Thing</option>
     <option value="thingdev">Sparkfun ESP8266 Thing - Dev Board</option>
   </optgroup>
+  <optgroup label="ESP32C3">
```
Comment (balloob) [inline src/wizard/boards.ts:24]:
```suggestion
  <optgroup label="ESP32-C3">
```

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
I was talking to @jesserockz and he mentioned that we use [`v3.5.0`](https://github.com/esphome/esphome/blob/dev/esphome/components/esp32/__init__.py#L136) and so that board name does not work with ESPHome.

**Comment by jesserockz** [comment]
More specifically, until the ESPHome core code is upgraded, there is no point in adding these to the dashboard UI as it will just cause issues for people who choose it.

**Comment by lakshanthad** [comment]
I understand. When will you upgrade the ESPHome core code with the latest version?

**Pull request closed without merging**